playbooks/roles/logstash/templates/logstash.conf.j2

```diff
@@ -1,43 +1,196 @@
 input {
-  tcp {
-    port => {{ logstash_syslog_port }}
-    type => syslog
-  }
-  udp {
-    port => {{ logstash_syslog_port }}
-    type => syslog
+  beats {
+    port => {{ LOGSTASH_PORT }}
+    type => beats
+    congestion_threshold => 40
   }
 }
 
 filter {
-  if [type] == "syslog" {
-    grok {
-      match => { "message" => "<%{POSINT:syslog_pri}>%{SYSLOGTIMESTAMP:syslog_timestamp} %{SYSLOGHOST:syslog_hostname} %{GREEDYDATA:syslog_message}" }
-      add_field => [ "received_at", "%{@timestamp}" ]
-      add_field => [ "received_from", "%{@source_host}" ]
-    }
-    syslog_pri { }
-    date {
-      match => [ "syslog_timestamp", "MMM  d HH:mm:ss", "MMM dd HH:mm:ss" ]
-    }
-    # Try and parse the tracking log json
-    # 142 is syslog facility 17 (local1) and Informational.
-    # This is used to reduce the number of errors in json parsing as
-    # tracking uses that facility and priority by default.
-    if "142" in [syslog_pri] {
-      json {
-        source => "syslog_message"
-        target => "tracking"
-      }
-    }
-    if !("_grokparsefailure" in [tags]) {
-      mutate {
-        replace => [ "@source_host", "%{syslog_hostname}" ]
-        replace => [ "@message", "%{syslog_message}" ]
-      }
+  multiline {
+    pattern => "(^(?![0-9]|[A-Z][a-z][a-z]\s|Traceback|\[[0-9]|date)|^([0-9][0-9]\s))"
+    what => "previous"
+    stream_identity => "%{source}"
+  }
+  if !([message] =~ /^(Traceback)/ ) {
+  if ("/edx/var/log/nginx/kibana.access.log" in [source]) {
+    grok {
+      match => { "message" => "%{IP:ip_addr} \- \- \[%{MONTHDAY}\/%{MONTH}\/%{YEAR}\:%{HOUR}\:%{MINUTE}\:%{SECOND} %{ISO8601_TIMEZONE}\] %{QUOTEDSTRING:nginx_request} %{INT:nginx_status_code} %{INT:nginx_payload_size} %{QUOTEDSTRING:nginx_external_url}%{GREEDYDATA:nginx_user_agent}" }
+    }
+  } else if ("/edx/app/edxapp/edx-platform/logs/app1/nginx/carnegie_access.log" in [source] or
```

Comment:
Some other examples: https://www.digitalocean.com/community/tutorials/adding-logstash-filters-to-improve-centralized-logging